org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/server/formatting/FormattingService.xtend

```diff
@@ -60,12 +68,32 @@ class FormattingService {
 	) {
 		val offset = document.getOffSet(params.range.start)
 		val length = document.getOffSet(params.range.end) - offset
-		return format(resource, document, offset, length)
+		if (OverrideChecker.hasFormatOverride(class)) {
+			return format(resource, document, offset, length)
+		}
+		return format(resource, document, offset, length, params.options)
 	}
-
+	
+	/**
+	 * @deprecated use {@link #format(XtextResource, document, offset, length, FormattingOptions} instead.
+	 */
+	@Deprecated
 	def List<TextEdit> format(XtextResource resource, Document document, int offset, int length) {
+		format(resource, document, offset, length, null)
+	}
+
+	/**
+	 * @since 2.14
+	 */
+	def List<TextEdit> format(XtextResource resource, Document document, int offset, int length, FormattingOptions options) {
+		var indent = "\t"
+		if (options !== null) {
+			if (options.insertSpaces) {
+				indent = Strings.padEnd("", options.tabSize," ")
+			}
+		}
 		if (formatter2Provider !== null) {
-			val replacements = format2(resource, new TextRegion(offset, length), null)
+			val replacements = format2(resource, new TextRegion(offset, length), new MapBasedPreferenceValues(#{"indentation"->indent}))
```

Comment:
Why do you create an immutable map here? 
Makes my code fail when trying to add additional values.

Reply:
where do you try to add additional stuff?

Reply:
besides this here is no intention behind this. so feel free to change